### PR TITLE
created crud UserServiceJpa and some fix in Message class and main app class

### PR DESCRIPTION
**Pull request by fac30ff**
Add UserServiceJpa class
add Temporal time stamp in message class
add jpa auditing for temporal timestamp in main class

### CODE REVIEW

**Comment by fac30ff** [comment]
[create USerServiceJPA class implemented UserService](https://trello.com/c/9ihGpx2J/68-create-userservicejpa-class-implemented-userservice)

File: src/main/java/com/textmessenger/model/Message.java [hunk 1]
```diff
@@ -30,5 +36,7 @@
   private String message;
 
   @Column(name = "created_date")
+  @Temporal(TemporalType.TIMESTAMP)
+  @CreatedDate
```
Comment (podolsky4) [inline src/main/java/com/textmessenger/model/Message.java]: Don't use java.utils.Date,use java.time.LocalDateTime instead

File: src/main/java/com/textmessenger/model/Message.java [hunk 2]
```diff
@@ -1,18 +1,24 @@
...
 import javax.persistence.Id;
 import javax.persistence.Table;
+import javax.persistence.Temporal;
+import javax.persistence.TemporalType;
 import java.util.Date;
 
 @Entity
 @Table(name = "messages")
 @Data
+@EntityListeners(AuditingEntityListener.class)
```
Comment (podolsky4) [inline src/main/java/com/textmessenger/model/Message.java]: What is this?

File: src/main/java/com/textmessenger/service/UserServiceJpa.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import com.textmessenger.model.User;
+import com.textmessenger.repository.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class UserServiceJpa implements UserService {
+
+  final UserRepository userRepository;
```
Comment (podolsky4) [inline src/main/java/com/textmessenger/service/UserServiceJpa.java]: private modifier

Comment (podolsky4) [reply src/main/java/com/textmessenger/service/UserServiceJpa.java]: autowired?

File: src/main/java/com/textmessenger/model/Message.java (hunk 2 above)
Comment (fac30ff) [reply src/main/java/com/textmessenger/model/Message.java]: Entity listener fro listening entity :D

File: src/main/java/com/textmessenger/model/Message.java (hunk 1 above)
Comment (fac30ff) [reply src/main/java/com/textmessenger/model/Message.java]: there are some reliable sources?

File: src/main/java/com/textmessenger/service/UserServiceJpa.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
+package com.textmessenger.service;
+
+import com.textmessenger.model.User;
+import com.textmessenger.repository.UserRepository;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@Transactional
+public class UserServiceJpa implements UserService {
```
[the remark this answers is not in the document]
Comment (fac30ff) [reply src/main/java/com/textmessenger/service/UserServiceJpa.java]: UserService is Interface ?! this is one of type, any suggestion?

File: src/main/java/com/textmessenger/model/Message.java (hunk 1 above)
Comment (zozich) [reply src/main/java/com/textmessenger/model/Message.java]: It's okay to use both

File: src/main/java/com/textmessenger/service/UserServiceJpa.java (hunk 4 above)
[the remark this answers is not in the document]
Comment (zozich) [reply src/main/java/com/textmessenger/service/UserServiceJpa.java]: `UserServiceImpl`

[the remark this answers is not in the document]
Comment (podolsky4) [reply src/main/java/com/textmessenger/service/UserServiceJpa.java]: Agreed

**Pull request merged**